src/Policies.cpp

```diff
@@ -299,7 +312,10 @@ void PolicyManager::remove_policy(const std::string &name)
     for (const auto &name : module_names) {
         _registry->handler_manager()->module_remove(name);
     }
-    _registry->input_manager()->module_remove(input_name);
+
+    if (policy->input_stream()->policies_count()) {
```

Comment:
Isn't this the opposite of what we want - shouldn't it be `!policy->input_stream()->policies_count()`? In other words, we only want to remove the module when there are no more policies...

If that's true, there must not be a test case for this, let's make one.

Reply:
yes. I missed the "!" here